src/libtriton/includes/triton/dllexport.hpp

```diff
@@ -15,6 +15,8 @@
     #else
       #define TRITON_EXPORT __declspec(dllexport) // Note: actually gcc seems to also supports this syntax.
     #endif
+  #elif BUILDING_LIB
```

Comment:
Who defined `BUILDING_LIB`?

Reply:
I have to define it in my app if I am importing a `.lib` instead of a `.dll` in Windows.

Reply:
shouldn't it be defined by the cmakelist ?

Reply:
Mmmm... I guess it will work too if you define it in the CMakeList.txt. Give me a minute and I'll test it

Reply:
The important part is defining it in the target app (the one who is importing the static library). there must be no prefix/decoration if importing from a static library so it should be:
`#define TRITON_EXPORT`

Anyway, I know that `__declspec(dllexport)` is necessary in Windows but, I am not sure that `__declspec(dllimport)` is so maybe the line:
`#define TRITON_EXPORT __declspec(dllimport) // Note: actually gcc seems to also supports this syntax.` can be replace by #define TRITON_EXPORT

Reply:
that will work for static libraries. Not sure about using `libtriton` as a dll